src/connector/src/sink/mod.rs

```diff
@@ -297,6 +297,17 @@ pub trait Sink: TryFrom<SinkParam, Error = SinkError> {
 
     async fn validate(&self) -> Result<()>;
     async fn new_log_sinker(&self, writer_param: SinkWriterParam) -> Result<Self::LogSinker>;
+    #[cfg(feature = "sink_bench")]
```

Comment:
IIUC, we introduce this new method because in benchmark we are not able to create a grpc meta client. If so, instead of adding a cargo feature and a new method in the sink trait, we can change the `meta_client` in `SinkWriterParam` to an enum that can be either grpc meta client, or our mocked meta client.

The call path of meta client in sink coordination is 
```
let sink_coordinate_client = meta_client.sink_coordinate_client();
let stream_handle = StreamHandle::new(sink_coordinate_client);
```
only the `stream_handle` at the end will be stored as a field in some sink writer. The `BidiStreamHandle` contains only a request channel sender and a box response stream, which can be easily mocked.

Therefore, we can have enum for `meta_client` and `coordinate_client`. The `meta_client` enum has method `fn sink_coordinate_client(&self)` to create the corresponding `sink_coordinate_client` enum. The `sink_coordinate_client` enum has method `fn new_stream_handle(self, PbSinkParam, Bitmap)` to create the stream handle. The grpc variant of the `sink_coordinate_client` enum will follow the original call path, while the mock variant will create a mocked bidi stream handle. 

Besides, since we only bench the performance of writer, we don't really need to create the coordinator to commit the metadata. When creating the mocked bidi stream handle, we can just spawn a tokio task to receive the commit request and return with a commit response immediately.

In this way, we will not have to have a new cargo feature and introduce so many hacks in the concrete sink implementation.